mirar/processors/photcal/zp_calculator/color_term_zp_calculator.py

```diff
@@ -51,7 +51,7 @@ def calculate_zeropoint(
         colors = matched_ref_cat[color_colnames[0]] - matched_ref_cat[color_colnames[1]]
 
         for colname in colnames:
-            y = matched_img_cat[colname] - matched_ref_cat["magnitude"]
+            y = matched_ref_cat["magnitude"] - matched_img_cat[colname]
             x = colors
             y_err = np.sqrt(
                 matched_img_cat[colname.replace("MAG", "MAGERR")] ** 2
```

Comment:
`matched_ref_cat["magnitude_err"]` also raises a KeyError. I think that's a column that only shows up in sourcetables? In any case, wouldn't this column name need to be specific to the reference catalog you are using?

I suggest: the generator function `self.color_colnames_generator` returns two things: `color_colnames, mag_err_colname` so for example, with panstarrs + g-band image you get `['gmag', 'rmag'], 'e_gmag'`

Later, if we want to use errors on the x and y axes, the generator may return `['gmag', 'rmag'], ['e_gmag', 'e_rmag']` (for the panstarrs gmag case)

Reply:
Done to both